### DISCUSSION TITLE
Jest not transforming react-markdown when using the jsdom environment React

### DISCUSSION THREAD

**Discussion by zheey** [remarkjs/remark#883] [Q&A] [answered]
Hi all,

Has anyone encountered this issue?
`Details:

    import {ReactMarkdown} from './lib/react-markdown.js'
    ^^^^^^

    SyntaxError: Cannot use import statement outside a module`
 I saw the a similar answer but doesn't seem to solve the issue.

Please, I will like to know how to resolve that.

**Comment by ChristianMurphy** [comment]
It is noted in the release notes https://github.com/remarkjs/react-markdown/blob/main/changelog.md#700---2021-08-13

Jest needs to be configured to support ESM, their documentation provides a guide on how to do this https://jestjs.io/docs/ecmascript-modules

**Comment by nvh95** [comment, marked as answer]
@zheey I encountered this issue earlier today and resolved it. You can see my suggestions here https://github.com/remarkjs/react-markdown/issues/635#issuecomment-956158474
(I picked option 3)

**Comment by darkocejkov** [comment]
just providing some future input, i've made a [gist that works](https://gist.github.com/darkocejkov/48b9394fbfe86d9a4154d6a102ce2caa) for me since the old answer wasn't directly working. the secret i found is to run the test, check what the next ESM module is failing then add that to the list.
